CoreFoundation/Base.subproj/CFRuntime.c

```diff
@@ -1817,6 +1818,28 @@ void * objc_retainAutoreleasedReturnValue(void *obj) {
     else return NULL;
 }
 
+// temporary helpers for poking into Swift runtime, for implementing NSStringFromClass()
+// needs to be replaced with a proper contract, this is fragile in so many ways
+
+typedef struct NominalTypeDescriptor {
+    uintptr_t Kind;
+    const char *Name;
+} NominalTypeDescriptor;
+
+extern const NominalTypeDescriptor *_ZNK5swift8Metadata24getNominalTypeDescriptorEv(const void *);
+
+_Nullable CFStringRef _CFCopyNominalTypeNameForClass(_Nonnull CFTypeRef aClass) {
+    const NominalTypeDescriptor *ntd;
+
+    CFAssert1(aClass != NULL, __kCFLogAssertion, "%s(): invalid class", __PRETTY_FUNCTION__);
+
+    ntd = _ZNK5swift8Metadata24getNominalTypeDescriptorEv(aClass);
```

Comment:
We can remove this but it will not be possible to encode >2 level classes (e.g. Foo.Bar.SomeClass). We could also mangle the name ourselves but it would be better to have a proper contract with the Swift runtime for transforming classes to names and the converse. If we do not care about interoperability with Darwin archives then such a putative API does not need to expose mangled names.